lib/Launcher.js

```
@@ -86,12 +89,16 @@ class Launcher {
       chromeProcess.stderr.pipe(process.stderr);
     }
 
-    const waitForChromeToClose = new Promise(fulfill => {
+    const waitForChromeToClose = new Promise((fulfill, reject) => {
       chromeProcess.once('close', () => {
         // Cleanup as processes exit.
-        if (temporaryUserDataDir)
-          removeSync(temporaryUserDataDir);
-        fulfill();
+        if (temporaryUserDataDir) {
+          removeFolderAsync(temporaryUserDataDir)
+              .then(() => fulfill())
+              .catch(err => console.error(err));
```

why not just let this throw, like removeSync would have done.

---

@graingert to avoid `unhandledPromiseRejection`.

---

presumably people are going to `await Page.pdf();` and thus catch any errors

---

How's `Page.pdf` related to this code?

---

@aslushnikov it's the API that calls the download

---

@aslushnikov ah you're right. Edited comment